bmap-writer.cpp

```diff
@@ -123,34 +146,18 @@ bool isDeviceMounted(const std::string &device) {
     return false;
 }
 
-void printBufferHex(const char *buffer, size_t size) {
-#ifdef DEBUG
-    for (size_t i = 0; i < size; ++i) {
-        std::cout << std::hex << std::setw(2) << std::setfill('0') << (unsigned int)(unsigned char)buffer[i];
-        if ((i + 1) % 16 == 0) {
-            std::cout << std::endl;
-        } else {
-            std::cout << " ";
-        }
-    }
-    std::cout << std::endl;
-#else
-    (void)buffer;
-    (void)size;
-    return;
-#endif
-}
-
-int BmapWriteImage(const std::string &imageFile, const bmap_t &bmap, const std::string &device) {
+int BmapWriteImage(const std::string &imageFile, const bmap_t &bmap, const std::string &device,
+                   const size_t maxBufferSize = 0, const enum ChecksumMode checksumMode = CHECKSUM_READ) {
     static const size_t read_block_size = 16384;
     struct archive *a = nullptr;
     int dev_fd = -1;
     int ret = 0;
 
     try {
         size_t decHead = 0;
+        int rwMode = (checksumMode == CHECKSUM_WRITTEN) ? O_RDWR : O_WRONLY;
```

Comment:
I think we can open it always in RW mode there should be no major drawbacks on performance

Reply:
Ok